src/main/java/org/opensearch/neuralsearch/processor/TextEmbeddingProcessor.java

```diff
@@ -0,0 +1,278 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.neuralsearch.processor;
+
+import static org.opensearch.ingest.ConfigurationUtils.readOptionalMap;
+import static org.opensearch.ingest.ConfigurationUtils.readStringProperty;
+
+import java.util.*;
+import java.util.function.Consumer;
+
+import lombok.extern.log4j.Log4j2;
+
+import org.apache.commons.lang3.StringUtils;
+import org.opensearch.action.ActionListener;
+import org.opensearch.client.Client;
+import org.opensearch.common.CheckedConsumer;
+import org.opensearch.ingest.AbstractProcessor;
+import org.opensearch.ingest.IngestDocument;
+import org.opensearch.ingest.Processor;
+import org.opensearch.ml.client.MachineLearningNodeClient;
+import org.opensearch.neuralsearch.ml.MLCommonsClientAccessor;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableMap;
+
+@Log4j2
+public class TextEmbeddingProcessor extends AbstractProcessor {
+
+    public static final String TYPE = "text_embedding";
+    public static final String MODEL_ID_FIELD = "model_id";
+    public static final String FIELD_MAP_FIELD = "field_map";
+
+    private static final String LIST_TYPE_NESTED_MAP_KEY = "knn";
+
+    private final String modelId;
+
+    private final Map<String, Object> fieldMap;
+
+    private final MLCommonsClientAccessor mlCommonsClientAccessor;
+
+    public TextEmbeddingProcessor(String tag, String description, String modelId, Map<String, Object> fieldMap, Client client) {
+        super(tag, description);
+        this.modelId = Objects.requireNonNull(modelId, "model_id is null, can not process it");
+        if (fieldMap == null || fieldMap.size() == 0 || checkEmbeddingConfigNotValid(fieldMap)) {
+            throw new IllegalArgumentException("filed_map is null, can not process it");
+        } else {
+            this.fieldMap = fieldMap;
+        }
+        this.mlCommonsClientAccessor = new MLCommonsClientAccessor(new MachineLearningNodeClient(client));
```

Comment:
New Class is not a best practice indeed, but `Processors` are created by `Factory` instead of injection, also the `NeuralSearch` plugin needs an implementation of `IngestionPlugin`, check below code in `NeuralSearch`:
```
@Override
    public Map<String, Processor.Factory> getProcessors(Processor.Parameters parameters) {
        return Collections.singletonMap(TextEmbeddingProcessor.TYPE, new TextEmbeddingProcessor.Factory(parameters.client));
    }
```
We need to return a factory map, and the instance creation happens out of our code by invoking the factory.create automatically, and a `@inject` field won't be initialized correctly.

Reply:
even with the 

```
@Override
    public Map<String, Processor.Factory> getProcessors(Processor.Parameters parameters) {
        return Collections.singletonMap(TextEmbeddingProcessor.TYPE, new TextEmbeddingProcessor.Factory(parameters.client));
    }
```

You can use the same instance that we have created using createComponents. I would say rather than passing client in the `new TextEmbeddingProcessor.Factory(parameters.client)` pass the MLCommonsAccessor.